keps/sig-windows/windows-runtimeclass-support.md

```diff
@@ -333,6 +336,46 @@ message PullImageRequest {
 }
 ```
 
+#### Adding annotations to ImageSpec
+
+Current proposal is to change the `ImageSpec` to incorporate annotations for runtime class. These can be passed down to the container runtimes where various actions can be taken as per the runtime class specified.
+ 
+Currently ImageSpec contains just a string.
+
+```
+type ImageSpec struct {
+  Image string
+}
+```
+ 
+The proposal is to add Annotations into the ImageSpec:
+
+```
+type ImageSpec struct {
+  Image string
+  Annotations []Annotation
+}
+```
+ 
+The runtimeHandler annotation will be based on the Runtime Class specified by the user:
+```
+“runtimehandler”: “<corresponding values>”
+ ```
+
+We could potentially also add the kubernetes specification annotations for consideration of the runtime:
+```
+“kubernetes.io/arch”: “amd64”
+“kubernetes.io/os”: ”linux”
```

Comment:
Should we define these in the `kubernetes.io/image/*` namespace?

Reply:
And have:
`.../arch`
`.../os`
`.../variant`
as known values?

Reply:
Ah good point. I see why this is confusing. Today we already have well-known labels applied to the node by the kubelet [doc](https://github.com/kubernetes/enhancements/blob/f1a799d5f4658ed29797c1fb9ceb7a4d0f538e93/keps/sig-auth/0000-20170814-bounding-self-labeling-kubelets.md#proposal)

* `kubernetes.io/os`
* `kubernetes.io/arch`

Those are intended to be used for scheduling as nodeSelectors, not for determining what image to pull.

In this KEP I propose to encode a user's _desired_ os/arch as part of the RuntimeClass. This may be different from the node labels above, so it would be good to use different names. RuntimeClass is the only thing passed in the pod sandbox config. I think for image/* to be useful, we'd need matching ones in the pod sandbox config too.

Reply:
Ah I see. Ok then yes we will just have one known annotation `kubernetes.io/runtimehandler` that is passed in the `ImageSpec.Annotations`. Thanks